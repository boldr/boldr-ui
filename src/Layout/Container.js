// @flow
import React from 'react';
import classNames from 'classnames';

import { createWrappedComponent } from '../util/boldrui';

export type Props = {
  tag?: string,
  isFluid?: boolean,
  className?: string,
};

export function Container({ tag = 'div', ...props }: Props) {
  const className = classNames(
    'boldrui-container',
    {
      'is-fluid': props.isFluid,
      'is-flex': true,
    },
    props.className,
  );

  const { ...HTMLProps } = props;

  return React.createElement(tag, { ...HTMLProps, className });
}

export default createWrappedComponent(Container);
