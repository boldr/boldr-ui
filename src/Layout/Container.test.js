import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Container } from './Container';

describe('Container', () => {
  // it('should render a div with .boldr-container', () => {
  //   const container = shallow(<Container>My Container</Container>);
  //   expect(container.contains(<div className="boldr-container">My Container</div>)).toBe(true);
  // });
  //
  // it('should render a p with .boldr-container', () => {
  //   const container = shallow(<Container tag="p">My Container</Container>);
  //   expect(container.contains(<p className="boldr-container">My Container</p>)).toBe(true);
  // });

  it('should render a div with .boldr-container with modifiers', () => {
    const container = shallow(<Container isFluid />);
    expect(container.hasClass('boldr-container')).toBe(true);
    expect(container.hasClass('is-fluid')).toBe(true);
  });

  it('should render a div with .boldr-container, modifiers and custom classNames', () => {
    const container = shallow(<Container isFluid className="custom loader" />);
    expect(container.hasClass('boldr-container')).toBe(true);
    expect(container.hasClass('is-fluid')).toBe(true);
    expect(container.hasClass('custom')).toBe(true);
    expect(container.hasClass('loader')).toBe(true);
  });

  it('should render a div with .boldr-container and custom classNames', () => {
    const container = shallow(<Container className="custom loader" />);
    expect(container.hasClass('boldr-container')).toBe(true);
    expect(container.hasClass('custom')).toBe(true);
    expect(container.hasClass('loader')).toBe(true);
  });
  it('should match the snapshot', () => {
    const wrapper = shallow(<Container>My Container</Container>);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
