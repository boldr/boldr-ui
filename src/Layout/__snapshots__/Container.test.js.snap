// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should match the snapshot 1`] = `
<div
  className="boldr-container is-flex"
>
  My Container
</div>
`;
